extends Node2D

# Dibuja tipo un diagrama de Venn que representa quÃ© tan alineadas
# estÃ¡n entre sÃ­ las "burbujas" que estÃ©n seteadas. Cada burbuja es
# una cantidad de sÃ­mbolos de los disponibles de la lista de sÃ­mbolos,
# y cada burbuja estarÃ­a representando un personaje.

var simbolos = ['ğŸ“¯', 'ğŸ“¿', 'ğŸª¢']
var colores = [Color.BROWN, Color.DARK_ORANGE, Color.LAWN_GREEN]
#var burbujas = ['ğŸ“¯ğŸ“¯ğŸ“¯', 'ğŸ“¿ğŸª¢ğŸ“¿', 'ğŸª¢ğŸª¢ğŸ“¿']
var burbujas = ['ğŸ“¯ğŸª¢ğŸª¢', 'ğŸ“¯ğŸª¢ğŸ“¿', 'ğŸ“¯ğŸª¢']
var polos = [] 

var default_font = ThemeDB.fallback_font
var default_font_size = 48
	
var diametro = 150

func _ready():
	update()

func update():
	# Calculamos las posiciones de los "polos"
	polos = []
	for simbolo in simbolos:
		var a = simbolos.find(simbolo) * TAU / simbolos.size()
		polos.append(diametro * Vector2.from_angle(a))

func posiciones():
	
	var ps = [] 
	
	for burbuja in burbujas:
		var pos = Vector2.ZERO
		for simbolo in burbuja: 
			pos += polos[simbolos.find(simbolo)] / 3
		ps.append(pos)
	
	return ps

func _draw():
	draw_circle(Vector2.ZERO, 140, Color(Color.GRAY, 0.4))

	# Dibujamos cada cÃ­rculo en su posiciÃ³n
	var centros = posiciones()
	for i in centros.size():
		draw_circle(centros[i], 80, Color(colores[i], 0.4))
		
	# Dibujamos los sÃ­mbolos
	for i in simbolos.size():
		draw_circle(polos[i], 35, Color(Color.WHITE,0.2))
		draw_string(default_font, polos[i] + Vector2(-24, 24), simbolos[i], HORIZONTAL_ALIGNMENT_LEFT, -1, default_font_size)
	

	
